packages/autocertifier/src/DnsServer.ts

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import { DnsHandler, DnsResponse, Packet, createServer } from 'dns2'
+import { Database, Subdomain } from './Database'
+import { Logger } from '@streamr/utils'
+
+const logger = new Logger(module)
+
+// TODO: there is appears to be a general problem with typing in this class. Seems that the DNS2 library does
+// not provide proper typing for many fields and response types. Alternativel we should just simply send all
+// responses as the DnsResponse type instead of unknown
+export class DnsServer {
+
+    private server?: any
+    private readonly domainName: string
+    private readonly ownHostName: string
+    private readonly dnsServerPort: string
+    private readonly ownIpAddress: string
+    private readonly db: Database
+
+    constructor(domainName: string, ownHostName: string, dnsServerPort: string, ownIpAddress: string, db: Database) {
+        this.domainName = domainName
+        this.ownHostName = ownHostName
+        this.dnsServerPort = dnsServerPort
+        this.ownIpAddress = ownIpAddress
+        this.db = db
+    }
+
+    public async createSubdomain(subdomain: string, ipAddress: string, port: string, token: string): Promise<void> {
+        await this.db.createSubdomain(subdomain, ipAddress, port, token)
+    }
+
+    public async updateSubdomainIpAndPort(subdomain: string, ipAddress: string, port: string, token: string): Promise<void> {
+        await this.db.updateSubdomainIpAndPort(subdomain, ipAddress, port, token)
+    }
+
+    public async updateSubdomainAcmeChallenge(fqdn: string, acmeChallenge: string): Promise<void> {
+        const parts = fqdn.split('.')
+        await this.db.updateSubdomainAcmeChallenge(parts[0], acmeChallenge)
+    }
+
+    private handleSOAQuery = async (mixedCaseName: string, send: (response: DnsResponse) => void,
+        response: DnsResponse): Promise<void> => {
+        // @ts-ignore private field
+        response.answers.push({
+            name: mixedCaseName,
+            type: Packet.TYPE.SOA,
+            class: Packet.CLASS.IN,
+            ttl: 86400,
+            primary: this.ownHostName + '.' + this.domainName,
+            admin: 'admin.' + this.domainName,
+            serial: 1,
+            refresh: 86400,
+            retry: 7200,
+            expiration: 3600000,
+            minimum: 172800,
+        } as unknown)
+        await send(response)
```

Comment:
```suggestion
        send(response)
```

Also some other invocations in this class